xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -368,17 +369,18 @@ where
             id: ref msg_id,
             ..
         } = *envelope;
+        let mut locked_openmls_group = openmls_group.lock().await;
```

Comment:
kinda we tried this idea in the beginning to see if that fixes the issue, but only as a POC.
the downside here is that other developers easily can forget to lock the group!

Reply:
> the downside here is that other developers easily can forget to lock the group!

Not sure what you mean. That would be the case with any implementation? You have to remember to lock the group no matter how you implement it. The upside of this approach is you can have functions require the `SerialOpenMlsGroup` as a param where locking is critical, so it's actually the opposite that would be true.